### PR TITLE
Allows the ID computer to save and apply custom jobs to ID cards

### PR DESCRIPTION
**Pull request by 112358sam**
[QoL]

## About the PR 
Adds three slots for custom jobs on the menu of the ID card computer. Each saves the job name and access list and can apply it to other cards with the press of a button. Save a job by setting the name and access and hitting save then it can be applied to future cards.

image

## Why's this needed? 
It was requested on the forums https://forum.ss13.co/showthread.php?tid=16609 
It saves the HoP time when trying to give the same access level out multiple times or lets the HoP easily build and distribute custom jobs and access levels.

## Changelog

```changelog
(u)112358sam
(+)Adds custom job save slots to the ID card computer.
```

### CODE REVIEW

**Comment by Awfullyfishy** [comment]
Is it possible to save the access and name of an ID to the slot? Meaning you could copy the HoS special access?

**Comment by 112358sam** [comment]
I didn't realize HOS had the unique access. I can remove it if it is an issue that it could be copied. Copying it would require having their card in the first place, but I can add a check to prevent it once I am back at my computer.

**Comment by Crackaduck** [comment]
> I didn't realize HOS had the unique access. I can remove it if it is an issue that it could be copied. Copying it would require having their card in the first place, but I can add a check to prevent it once I am back at my computer.

idk I think its fine, the only way it comes into play is if hos willingly gives it away to someone else. You'd have to have the Hos's card to do it without their consent in the first place, at which point, youve probably already killed them and taken the id for yourself. Besides that I think it actually adds flavor, the only way to get hos's acess on the computer is to get the card from them. TLDR its not a bug its a feature :)

**Comment by 112358sam** [comment]
I went ahead and removed the ability to store the HoS special access. I can easily revert this if we want to go the feature route.

**Comment by Crackaduck** [comment]
> I went ahead and removed the ability to store the HoS special access. I can easily revert this if we want to go the feature route.

can it still store their office or are they one in the same?

**Comment by 112358sam** [comment]
At least on cog 2 where I tested it, the armory access and HoS office use the same access.

**Comment by nefarious6th** [comment]
armory and office access are tied together, yea.

**Comment by ThatFiveGuys** [comment]
Make sure it cant copy listening post access from agent id cards too.

**Comment by 112358sam** [comment]
> 
> 
> Make sure it cant copy listening post access from agent id cards too.

I've updated the limit to block all azone accesses and syndicate outpost from being copied

**Comment by Crackaduck** [comment]
I disagree with it being unable to save certain locations access, as I'd like it better if it could save and access and save ALL access areas, but I understand for balance reasons why that could be bad (I think it would be fair but clearly im in the minority). should be a fun PR regardless, and good luck on your next projects!

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: code/obj/machinery/computer/card.dm [hunk 1]
```diff
@@ -83,7 +90,7 @@
 			var/list/civilianjobs = list("Staff Assistant", "Bartender", "Chef", "Botanist", "Rancher", "Chaplain", "Janitor", "Clown")
 			var/list/maintainencejobs = list("Engineer", "Mechanic", "Miner", "Quartermaster")
 			var/list/researchjobs = list("Scientist", "Medical Doctor", "Geneticist", "Roboticist", "Pathologist")
-			var/list/securityjobs = list("Security Officer", "Detective")
+			var/list/securityjobs = list("Security Officer", "Security Assistant", "Detective")
```
[the remark this answers is not in the document]
Comment (112358sam) [reply code/obj/machinery/computer/card.dm]: Removed and PR'd separately. #5485

File: code/obj/machinery/computer/card.dm [hunk 2]
```diff
@@ -334,6 +346,40 @@
...
+		var/slot = href_list["apply"]
+		if (slot == "custom1")
+			src.modify.assignment = src.custom1_name
+			src.modify.access = src.custom1_list.Copy()
+		else if (slot == "custom2")
+			src.modify.assignment = src.custom2_name
+			src.modify.access = src.custom2_list.Copy()
+		else
+			src.modify.assignment = src.custom3_name
+			src.modify.access = src.custom3_list.Copy()
```
[the remark this answers is not in the document]
Comment (112358sam) [reply code/obj/machinery/computer/card.dm]: Fixed!

File: code/obj/machinery/computer/card.dm [hunk 3]
```diff
@@ -107,6 +114,11 @@
 			for(var/job in commandjobs)
 				body += " <a href='?src=\ref[src];assign=[job];colour=green'>[replacetext(job, " ", "&nbsp")]</a>"
 
+			body += "<br>Custom:"
+			body += " [src.custom1_name] <a href='?src=\ref[src];save=custom1'>save</a> <a href='?src=\ref[src];apply=custom1'>apply</a>"
+			body += " [src.custom2_name] <a href='?src=\ref[src];save=custom2'>save</a> <a href='?src=\ref[src];apply=custom2'>apply</a>"
+			body += " [src.custom3_name] <a href='?src=\ref[src];save=custom3'>save</a> <a href='?src=\ref[src];apply=custom3'>apply</a>"
```
[the remark this answers is not in the document]
Comment (112358sam) [reply code/obj/machinery/computer/card.dm]: Done!

File: code/obj/machinery/computer/card.dm [hunk 4]
```diff
@@ -8,6 +8,13 @@
 	var/mode = 0.0
 	var/printing = null
 	var/list/scan_access = null
+	var/custom1_name = "Custom 1"
+	var/custom2_name = "Custom 2"
+	var/custom3_name = "Custom 3"
+	var/list/custom1_list = list()
+	var/list/custom2_list = list()
+	var/list/custom3_list = list()
+	var/list/banned_access_list = list(access_maxsec,access_syndicate_shuttle,access_owlerymaint,access_owlerysec,access_owlerycommand,access_syndicate_4,access_syndicate_8,access_syndicate_16,access_syndicate_32,access_syndicate_64,access_syndicate_128,access_syndicate_256,access_syndicate_512,access_polariscargo,access_polarisimportant,access_contrabandpermit,access_syndicate_commander,access_retention_blue,access_retention_green,access_retention_yellow,access_retention_orange,access_retention_red,access_retention_black)
```
[the remark this answers is not in the document]
Comment (112358sam) [reply code/obj/machinery/computer/card.dm]: Done!

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**